app/controllers/autocomplete_controller.rb

```diff
@@ -88,18 +88,40 @@ def associated_tags
 
   ## NONCANONICAL TAGS
   def noncanonical_tag
-    search_param = params[:term]
+    search_param = Query.new.escape_reserved_characters(params[:term])
     raise "Redshirt: Attempted to constantize invalid class initialize noncanonical_tag #{params[:type].classify}" unless Tag::TYPES.include?(params[:type].classify)
+
     tag_class = params[:type].classify.constantize
-    render_output(tag_class.by_popularity
-                      .where(["canonical = 0 AND name LIKE ?",
-                              '%' + search_param + '%']).limit(10).map(&:name))
+    one_tag = tag_class.find_by(canonical: false, name: params[:term])
+    # Is there a tag which is just right (this is really for testing)
```

Comment:
Could you explain how this is just for testing? I see `match` is used in rendering the output so I am slightly confused.

Reply:
@zz9pzza please correct me if I'm wrong, but this is hit in if a user types the whole tag into autocomplete without choosing one of the suggestions first. The tests definitely do that, but in theory a real user could as well (although that seems potentially unlikely)

Reply:
If this is used where I think it's used (the Synonyms field on canonical tags' edit pages), there's actually a pretty good choice the wrangler will be entering the name of the exact tag they're looking for, fwiw!

Reply:
The code will not find exact matches naturally, adding the check for the exact match match makes this use case work and the elasticsearch code finds good matches for the autocomplete.